sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -2549,6 +2551,47 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("Move the post-Scan Filters to the far right") {
+    val t1 = s"${catalogAndNamespace}table"
+    val sbq = "sbq"
+    withTable(t1) {
+      sql(s"CREATE TABLE $t1 (id bigint, data string) USING $v2Format")
+      sql(s"INSERT INTO $t1 VALUES (1, 'a'), (2, 'b'), (3, 'c')")
+
+      val filterBefore = spark.sql(
+        s"""
+           |SELECT id, data FROM $t1
+           |where md5(data) = '8cde774d6f7333752ed72cacddb05126'
```

Comment:
can we use a java UDF? I'm afraid we may support translating `md5` in the near future.

Reply:
> can we use a java UDF? I'm afraid we may support translating `md5` in the near future.

@cloud-fan Okay, i will optimize this

Reply:
> > can we use a java UDF? I'm afraid we may support translating `md5` in the near future.
> 
> @cloud-fan Okay, i will optimize this

@cloud-fan Updated